aodndata/soop/soop_tmv_nrt.py

```diff
@@ -134,28 +174,44 @@ def preprocess(self):
             raise RuntimeError(
                 "Missing vessel callsign {callsign} from vocabulary'.".format(callsign=SHIP_CODE))
 
-        f = self.file_collection[0]
+        f_txt = self.file_collection.filter_by_attribute_value('extension', '.txt')
+        f_log = self.file_collection.filter_by_attribute_value('extension', '.log')
+        f_nc = self.file_collection.filter_by_attribute_id('file_type', FileType.NETCDF)
 
-        if f.extension == '.nc':
+        if len(f_nc):
+            f_nc = f_nc[0]
             # case where we re-push an existing netcdf file
```

Comment:
not quite sure when we would re-push netcdf files?

Reply:
That can happen. to do some re-processing. I've done it already